test/presentation/bloc/movie_from_endpoint_bloc_test.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:movie_app_jsalinas/src/core/util/data_state.dart';
+import 'package:movie_app_jsalinas/src/core/util/enums.dart';
+import 'package:movie_app_jsalinas/src/domain/entity/genre.dart';
+import 'package:movie_app_jsalinas/src/domain/entity/movie.dart';
+import 'package:movie_app_jsalinas/src/domain/repository/i_movie_repository.dart';
+import 'package:movie_app_jsalinas/src/domain/usecase/implementation/fetch_movies_from_endpoint_usecase.dart';
+import 'package:movie_app_jsalinas/src/domain/usecase/implementation/fetch_valid_genres_usecase.dart';
+import 'package:movie_app_jsalinas/src/presentation/bloc/movie_from_endpoint_bloc.dart';
+
+class MockMovieRepository extends Mock implements IMovieRepository {}
+
+void main() {
+  group('calling bloc fetch methods', () {
+    final IMovieRepository repository = MockMovieRepository();
+
+    final MoviesFromEndpointBloc bloc = MoviesFromEndpointBloc(
+      fetchMoviesUseCase: FetchPopularUseCase(
+        repository: repository,
+      ),
+      fetchValidGenresUseCase: FetchValidGenresUseCase(
+        repository: repository,
+      ),
```

Comment:
Here you should mock the usecases instead, as you're only testing the bloc, not also the usecases.